### ISSUE TITLE
"make test" fails under Ubuntu 16.04

### ISSUE THREAD

**Issue by cambazz** [opencog/atomspace#1215]
Hello,

I have followed the instructions and managed to compile atomspace, but when I issued make test command I got the following error:

`[ 30%] Building CXX object tests/truthvalue/CMakeFiles/EvidenceCountTruthValueUTest.dir/EvidenceCountTruthValueUTest.cpp.o
In file included from /usr/include/cxxtest/RealDescriptions.h:20:0,
                 from /usr/include/cxxtest/TestRunner.h:22,
                 from /home/can/atomspace/build/tests/truthvalue/EvidenceCountTruthValueUTest.cpp:10:
/home/can/atomspace/build/tests/truthvalue/../../../tests/truthvalue/EvidenceCountTruthValueUTest.cxxtest: In member function ‘void EvidenceCountTruthValueUTest::test_mean()’:
/home/can/atomspace/build/tests/truthvalue/../../../tests/truthvalue/EvidenceCountTruthValueUTest.cxxtest:82:5: error: ‘isnan’ was not declared in this scope
     TS_ASSERT_IS_NAN(ectvs[i]->getMean());
     ^
/home/can/atomspace/build/tests/truthvalue/../../../tests/truthvalue/EvidenceCountTruthValueUTest.cxxtest:82:5: note: suggested alternative:
In file included from /usr/include/c++/5/complex:44:0,
                 from /usr/include/cxxtest/StdHeaders.h:24,
                 from /usr/include/cxxtest/StdValueTraits.h:22,
                 from /usr/include/cxxtest/TestSuite.h:25,
                 from /usr/include/cxxtest/RealDescriptions.h:20,
                 from /usr/include/cxxtest/TestRunner.h:22,
                 from /home/can/atomspace/build/tests/truthvalue/EvidenceCountTruthValueUTest.cpp:10:
/usr/include/c++/5/cmath:641:5: note:   ‘std::isnan’
     isnan(_Tp __x)
     ^
tests/truthvalue/CMakeFiles/EvidenceCountTruthValueUTest.dir/build.make:66: recipe for target 'tests/truthvalue/CMakeFiles/EvidenceCountTruthValueUTest.dir/EvidenceCountTruthValueUTest.cpp.o' failed
make[3]: *** [tests/truthvalue/CMakeFiles/EvidenceCountTruthValueUTest.dir/EvidenceCountTruthValueUTest.cpp.o] Error 1
CMakeFiles/Makefile2:2170: recipe for target 'tests/truthvalue/CMakeFiles/EvidenceCountTruthValueUTest.dir/all' failed
make[2]: *** [tests/truthvalue/CMakeFiles/EvidenceCountTruthValueUTest.dir/all] Error 2
CMakeFiles/Makefile2:106: recipe for target 'CMakeFiles/test.dir/rule' failed
make[1]: *** [CMakeFiles/test.dir/rule] Error 2
Makefile:197: recipe for target 'test' failed
make: *** [test] Error 2
`

Any ideas/recommendations greatly appreciated,

Best,
C.

**Comment by linas** [comment]
`isnan` is provided by the C library!  So that's a bizarre error. Perhaps add `#include <math.h>` will fix that.

**Comment by cambazz** [comment]
Hello Linas,

So I went ahead to edit the EvidenceCountTruthValueUTest.cxxtest file in:

/home/can/atomspace/tests/truthvalue/EvidenceCountTruthValueUTest.cxxtest 

And found out that it already has a 

#include <math.h>

#include <opencog/truthvalue/EvidenceCountTruthValue.h>
#include <opencog/util/Logger.h>
#include <opencog/util/exceptions.h>

Best regards,
C.

**Comment by jswiergo** [comment]
@cambazz could you provide information about your OS version, compiler version, cxxtest library version? How did you install cxxtest library? Using apt-get? Perhaps your cxxtest library is too old and does not work with new compiler.

**Comment by cambazz** [comment]
Hello,
I am on ubuntu 16.04 LTS, and I did everything with apt-get. cxxtest library is version 4.4

**Comment by ngeiswei** [comment]
I have seen this issue before, with Ubuntu 16.04, but I don't know what is wrong.

**Comment by jswiergo** [comment]
In cxxtest macro TS_ASSERT_IS_NAN expands to isnan() without std:: namespace prefix. It does not work with gcc 5.4.0 compiler (works e.g for 4.8.4) if isnan is used within a code using namespace. For example:
```
// compile: c++ -Wall -std=c++11 -o example Example.cc

#include <cstdio>
#include <math.h>

namespace opencog {};
using namespace opencog;

int main()
{
    printf("%d", isnan(0.0/0.0));
}
```
The snippet fails for gcc 5.4.0 (Ubuntu 16.04), works for gcc 4.8.4 (Ubuntu 14.10)

This is a bug in cxxtest. The workaround is to use namespace std by the tests that use macro TS_ASSERT_IS_NAN. I will send a patch.

**Comment by jswiergo** [comment]
Hmm. It seems Cxxtest project is not maintained by anybody since last Release 4.4 in 2014.

**Comment by linas** [comment]
I just now pushed a different fix in 236c0ab817a20794240c3ebf5c27046be30853f8

**Comment by linas** [comment]
closing, I assume this is fixed.
